### PR TITLE
add EntriesController and ReviewsController for HTTP method mappings

### PR DESCRIPTION
**Pull request by veropossumus**

### CODE REVIEW

File: src/main/java/com/peerrequest/app/api/EntriesController.java [hunk 1]
```diff
@@ -16,39 +13,50 @@
...
+    @PatchMapping(value = "categories/{categoryId}/entries/{entryId}",
+            consumes = "application/json",
+            produces = "application/json")
+    public Entry patchEntry(@PathVariable final int categoryId,
+                                    @PathVariable final int entryId,
+                                    @RequestBody final Entry.EntryUpdater entryUpdater) {
         Entry updatedEntry = mockEntry();
         updatedEntry.setAuthors("Karol Laseconde");
         updatedEntry.setTitle("Updated Title");
         return mockEntry();
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/api/EntriesController.java:53]: You set the author and title of updatedEntry and then return mockEntry?
Why?

File: src/main/java/com/peerrequest/app/api/EntriesController.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                                     @PathVariable final int entryId) {
+        return mockEntry().getDocument();
+    }
+
+    @PostMapping(value = "categories/{categoryId}/entries",
+            consumes = "application/json",
+            produces = "application/json")
+    public Entry postEntry(@PathVariable final int categoryId,
+                             @RequestBody final Entry entry) {
+        return mockEntry();
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/EntriesController.java:41]: should return the entry of the request body

File: src/main/java/com/peerrequest/app/api/EntriesController.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    @PatchMapping(value = "categories/{categoryId}/entries/{entryId}",
+            consumes = "application/json",
+            produces = "application/json")
+    public Entry patchEntry(@PathVariable final int categoryId,
+                                    @PathVariable final int entryId,
+                                    @RequestBody final Entry.EntryUpdater entryUpdater) {
+        Entry updatedEntry = mockEntry();
+        updatedEntry.setAuthors("Karol Laseconde");
+        updatedEntry.setTitle("Updated Title");
+        return updatedEntry;
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/EntriesController.java:53]: doesn't update the hardcoded entry. Maybe updating the hardcoded entry is not needed. Evaluate by yourself.

File: src/main/java/com/peerrequest/app/api/ReviewsController.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        return mockReview().getReviewDocument();
+    }
+
+    @PostMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews",
+            consumes = "application/json",
+            produces = "application/json")
+    public Review postReview(@PathVariable final int categoryId,
+                               @PathVariable final int entryId,
+                               @RequestBody final Review review) {
+        return mockReview();
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/ReviewsController.java:39]: Should return the review in the requestBody

File: src/main/java/com/peerrequest/app/api/ReviewsController.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @PutMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews/{reviewId}/document",
+            consumes = "application/json",
+            produces = "application/json")
+    public Document putReviewDocument(@PathVariable final int categoryId,
+                                    @PathVariable final int entryId,
+                                    @PathVariable final int reviewId,
+                                    @RequestBody final Document document) {
+        return mockReview().getReviewDocument();
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/ReviewsController.java:49]: should replace the reviewDocument of the hardcoded review

File: src/main/java/com/peerrequest/app/api/ReviewsController.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            consumes = "application/json",
+            produces = "application/json")
+    public Review patchReview(@PathVariable final int categoryId,
+                                      @PathVariable final int entryId,
+                                      @PathVariable final int reviewId,
+                                      @RequestBody final Review.ReviewUpdater reviewUpdater) {
+        Review updatedReview =  mockReview();
+        updatedReview.setScore(5);
+        updatedReview.setSummary("This is a brand new summary.");
+        return updatedReview;
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/ReviewsController.java:63]: doesn't update the hardcoded review. Maybe updating the hardcoded review is not needed. Evaluate by yourself.

**Review by Wil-Lee — CHANGES_REQUESTED** [review]
You find examples in BiddingProcessesController and RequestsController to fix the issues

**Comment by bits0rcerer** [comment]
Before merging we should wait for #28.

I think I will do #28 tomorrow and then we will see if our endpoints work as intended and return the correct JSON. :crossed_fingers:

**Review by Wil-Lee — DISMISSED** [review]

**Review by Wil-Lee — APPROVED** [review]

**Pull request merged**